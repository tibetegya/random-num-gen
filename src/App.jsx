import React, { useState, useEffect, useReducer } from 'react';
import './assets/styles/app.scss'
import yellowLine from './assets/images/yellow-line.svg';
import greenLine from './assets/images/green-line.svg';
import purpleCircle from './assets/images/purple-circle.svg';
import phoneMake from './assets/images/phonemake.png';
import Modal from './components/Modal';

const App = (props) => {
  const [modalOpen, setModalOpen] = useState(false);
  const [companyName, setCompanyName] = useState('');
  const [isSignedIn, setIsSignedIn] = useState(false);
  useEffect(() => {
    if(localStorage.getItem('companyName')){
      props.history.push('/dashboard');
    }
  }, [isSignedIn, props]);

  const handleSignIn = (e) => {
    const { target: { name, value } } = e;
    e.preventDefault();
    switch (name) {
      case 'companyName':
      setCompanyName(value);
        break;
      case 'signIn':
        localStorage.setItem('companyName', companyName)
        setIsSignedIn(true);
        setModalOpen(false);
        break;
      default:
    };

  }
  const handleModal = (e) => {
    const { target: { name } } = e;
    e.preventDefault();
    switch (name) {
      case 'closeModal':
        setModalOpen(false);
        break;
      case 'openModal':
        setModalOpen(true);
        break;
      default:
    };
  };
  return (
    <React.Fragment>
      <div className="wrapper" id="app">
      <img src={yellowLine} alt="line" className="bg" id="yellow-ln" />
      <img src={greenLine} alt="line" className="bg" id="green-ln" />
      <img src={purpleCircle} alt="line" className="bg" id="purple-cir" />
      <img src={phoneMake} alt="logo" className="logo"/>
      <div className="title">PhoneMake</div>
      <p className="description">
        PhoneMake is a service that powers telecom companies by <br/>
        generating unique phone numbers<br/><br/>
        Enter a company below in order to <br/>
        generate a collection of phone numbers<br/>
        </p>
        <button
          name="openModal"
          className="get-started btn"
          onClick={handleModal}
        >
        Get Started
        </button>
        <Modal
          handleModal={handleModal}
          isOpen={modalOpen}
          title="Get Started"
        >
          <h4>Create an account to get started</h4>
          <input
            type="text"
            name="companyName"
            id="company-name"
            placeholder="Enter company name"
            onChange={handleSignIn}
          />
          <button
            name="signIn"
            className="sign-in btn"
            onClick={handleSignIn}
          >
          Enter
          </button>
        </Modal>
      </div>
    </React.Fragment>
  );
};


export default App;
