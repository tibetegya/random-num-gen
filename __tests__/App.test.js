import React from 'react';
import { shallow, mount, render } from 'enzyme';

import App from '../src/App';

describe('App component test', () => {
  const mockFn = jest.fn();
  afterEach(() => {
    mockFn.mockClear()
  })

  it('should match snapshot', () => {
    const wrapper = mount(<App
      history={{
        push: mockFn,
      }}
    />);
    expect(wrapper).toMatchSnapshot();
  })

  it('should call handleModal method when a button is clicked', () => {
    const wrapper = shallow(<App
      history={{
        push: mockFn,
      }}
    />);
    const button = wrapper.find('button.get-started').simulate('click', {
      preventDefault: mockFn,
      target: {
        name:'username'
      }
    });
    expect(mockFn).toHaveBeenCalled();
  })
})