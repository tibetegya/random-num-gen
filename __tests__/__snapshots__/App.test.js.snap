// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App component test should match snapshot 1`] = `ReactWrapper {}`;
